pkg/cmd/init/exec.go

```diff
@@ -140,10 +149,29 @@ func (o *Options) run() error {
 	}
 	output = append(output, out...)
 
+	if o.wait && !o.ClusteradmFlags.DryRun {
+		if err := waitUntilClusterManagerRegistrationReady(
+			o.ClusteradmFlags.KubectlFactory,
+			int64(o.ClusteradmFlags.Timeout)); err != nil {
+			return err
+		}
+	}
+
 	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
 	if err != nil {
 		return nil
 	}
+	if len(o.outputJoinCommandFile) > 0 {
+		sh, err := os.OpenFile(o.outputJoinCommandFile, os.O_CREATE|os.O_WRONLY, 0755)
```

Comment:
Should have a flag to only output this command so we can pipe it to a file?

Reply:
are you thinking about utilizing the output from init e.g.?

> echo "$JOIN_CLUSTER_NAME" | cluster init --cmd-only --pipe-stdin